torchtune/recipe_interfaces.py

```diff
@@ -85,3 +85,36 @@ def evaluate(self, **kwargs) -> None:
         All of the evaluation logic, including reporting.
         """
         ...
+
+
+class OrchestrationRecipeInterface(Protocol):
```

Comment:
I'm not super keen on splitting orchestration out into it's own recipe. But I'm okay with adding this for now as that's what the current recipe looks like.

Reply:
I don't know if this is the best protocol for this, but it does make sense to me for this to be separate as you are not training but you are orchestrating, right?